client/client.go

```diff
@@ -69,28 +69,38 @@ func Marshal(c string, s interface{}) []string {
 	for i := 0; i < elem.NumField(); i++ {
 		value := elem.Field(i)
 		fieldType := elem.Type().Field(i)
-		// supports multiple struct tags--assumes first is mikrotik field name
-		tag := strings.Split(fieldType.Tag.Get("mikrotik"), ",")[0]
+		// fetch mikrotik struct tag, which supports multiple values separated by commas
+		tags := fieldType.Tag.Get("mikrotik")
+		// extract tag value that is the Mikrotik property name
+		// it is assumed that the first is mikrotik field name
+		mikrotikPropName := strings.Split(tags, ",")[0]
+
+		if mikrotikPropName != "" && (!value.IsZero() || value.Kind() == reflect.Bool) {
+			// add conditional to check if a Mikrotik property is READ ONLY, such as the following wireguard props
+			// https://help.mikrotik.com/docs/display/ROS/WireGuard#WireGuard-Read-onlyproperties
+			if strings.Contains(tags, "readonly") {
```

Comment:
I am fan of stricter checks, rather than using `substring`:
```suggestion
		mikrotikTags := strings.Split(tags, ",")
		mikrotikPropName := mikrotikTags[0]
		// now we have field name in separate variable,
		// so leave only modifiers in this slice
		mikrotikTags = mikrotikTags[1:]

		if mikrotikPropName != "" && (!value.IsZero() || value.Kind() == reflect.Bool) {
			// add conditional to check if a Mikrotik property is READ ONLY, such as the following wireguard props
			// https://help.mikrotik.com/docs/display/ROS/WireGuard#WireGuard-Read-onlyproperties
			if contains(mikrotikTags, "readonly") {


```
but it's just a matter of taste/confident, so feel free to merge the PR without this suggestion.

Looks totally good to me

Reply:
Sorry, was traveling the last few weeks. I agree with and like your suggestion -- I'll go ahead and make the change 👍